File: week-04/day-2/src/garden/Plant.java
package garden;

public class Plant {

    String color;
    double waterNeed;
    double waterAbsorbtion;

    public Plant(String color , int waterNeed, double waterAbsorbtion ){
        this.color = color;
        this.waterAbsorbtion = waterAbsorbtion;
    }

    public  void status(){
        if(waterNeed < 5){
            System.out.println("The " + color + " " + this.getClass() + " needs water.");
        } else {
            System.out.println("The " + color + " " + this.getClass() + " doesn't needs water.");
        }
    }
}

File: week-10/day-4/Garden/Garden/Plant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Garden
{
    abstract class Plant{

        protected String color;
        protected double waterNeed;
        protected double waterAbsorbtion;

        public Plant(String color, double waterNeed, double waterAbsorbtion) {
            this.color = color;
        }

        public abstract void Status();

        public void SetWaterNeed(double newWaterNeed) {
            this.waterNeed=newWaterNeed;
        }

        public double GetWaterNeed() {
            return waterNeed;
        }

        public double GetWaterAbsorption() {
            return waterAbsorbtion;
        }
    }

}
